x-pack/plugins/enterprise_search/public/applications/app_search/utils/format_api_name.test.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { formatApiName } from './format_api_name';
+
+describe('formatApiName', () => {
+  it('will format an api name', () => {
+    expect(formatApiName('SomeName')).toEqual('somename');
+  });
+
+  it('will replace sequences of non alpha-numeric characters with a dash', () => {
+    expect(formatApiName('*f1  &&o$ 1  2 *&%da ')).toEqual('f1-o-1-2-da');
+  });
```

Comment:
Would love to make the tests a little more deliberate based on the chained actions I'm seeing:

```suggestion
  it('replaces non-alphanumeric characters with dashes', () => {
    expect(formatApiName('f1  &&o$ 1  2 *&%da')).toEqual('f1-o-1-2-da');
  });

  it('strips leading and trailing non-alphanumeric characters ', () => {
    expect(formatApiName('$$hello world**')).toEqual('hello-world');
  });

  it('strips leading and trailing whitespace ', () => {
    expect(formatApiName('  test  ')).toEqual('test');
  });

  it('lowercases text ', () => {
    expect(formatApiName('SomeName')).toEqual('somename');
  });
```

I think adding extra specificity to the tests helps a lot with dev understanding - what do you think?

Reply:
Yes, agreed.

Reply:
4ff917d08ebadc121da81a6c53cc73a1f878234f